pkg/controller/controller.go

```diff
@@ -592,6 +592,20 @@ func (c *RolloutController) podsNotMatchingUpdateRevision(sts *v1.StatefulSet) (
 		return nil, err
 	}
 
+	// If there are no pods to be found, check if there are pods that need to be gracefully terminated.
+	if len(pods) == 0 {
```

Comment:
do we need this condition? don't we want to combine the pods which has graceful-shutdown label with existing pods list?

Reply:
No, because the precedence of the run is for it to enact on rolling out new changes. If all pods are updated, then check for the graceful-shutdown label.